doc/source/security/data-masking.rst

```diff
@@ -0,0 +1,307 @@
+sensitive data with modified content.
+
+Data masking can have either of the characteristics:
+
+* Generation of random data, such as an email address
+
+* De-identify data by transforming the data to hide content
+
+.. rubric:: Installing the plugin
+
+The following command installs the plugin:
+
+.. code-block:: bash
+
+    INSTALL PLUGIN data_masking SONAME 'data_masking.so';
+
+.. rubric:: Data Masking functions
+
+The data masking functions have the following categories:
+
+* General purpose
+
+* Special purpose
+
+* Generating Random Data with Defined characteristics
+
+* Using Dictionaries to Generate Random Data
+
+.. rubric:: General Purpose
+
+
+The general purpose data masking functions are the following:
+
+.. tabularcolumns:: |p{0.25\linewidth}|p{0.25\linewidth}|p{0.50\linewidth}|
+
+.. list-table::
+    :widths: 20 30 60
+    :header-rows: 1
+
+    * - Parameter
+      - Description
+      - Sample
+    * - mask_inner(string, margin1, margin2 [, character])
+      - Returns a result where only the inner part of a string is masked. An
+        optional masking character can be specified.
+      - .. code-block:: MySQL
+
+            mysql> SELECT mask_inner('123456789', 1, 2);
+
+            +-----------------------------------+
+            | mask_inner("123456789", 1, 2)     |
+            +-----------------------------------+
+            |1XXXXXX89                          |
+            +-----------------------------------+
+
+    * - mask_outer(string, margin1, margin2 [, character])
+      - Masks the outer part of the string. The inner section is not masked.
+      - .. code-block:: MySQL
+
+            mysql> SELECT mask_outer('123456789', 2, 2);
+
+            +------------------------------------+
+            | mask_outer("123456789", 2, 2).     |
+            +------------------------------------+
+            | XX34567XX                          |
+            +------------------------------------+
+
+.. rubric:: Special Purpose
+
+The special purpose data masking functions are as follows:
+
+.. tabularcolumns:: |p{0.25\linewidth}|p{0.25\linewidth}|p{0.50\linewidth}|
+
+.. list-table::
+    :widths: 20 30 60
+    :header-rows: 1
+
+    * - Parameter
+      - Description
+      - Sample
+
+    * - mask_pan(string)
+      - Masks the Primary Account Number (PAN) by replacing the
+        string with an "X" except for the last four characters.
+
+        .. note::
+
+            The PAN string must be 15 characters or 16 characters in length.
+
+      - .. code-block:: MySQL
+
+            mysql> SELECT mask_pan ('123456789012345');
+
+            +------------------------------------+
+            | mask_pan(gen_rnd_pan()).           |
+            +------------------------------------+
+            | XXXXXXXXXXX2345                    |
+            +------------------------------------+
+
+    * - mask_pan_relaxed(string)
+      - Returns the first six numbers and the last four    numbers. The rest of
```

Comment:
```suggestion
      - Returns the first six numbers and the last four numbers. The rest of
```

Reply:
Fixed.

Reply:
I still see multiple spaces between four and numbers..

Reply:
Odd. Not in raw doc.

Reply:
fixed.

Reply:
I still see the same :) May be you can use commit suggestion.

Reply:
Ok, I see it fixed now..